### PR TITLE
CP-24118: affinity settings, release notes

### PR DESCRIPTION
**Pull request by josephbarnett**
### Description

1. Updated insights controller to version 0.0.6
1. Changes to distribute pod replicas to different nodes (best effort)
2. Allow override of scrape target for KMS (DVAG beta feedback)
3. Release notes for cumulative changes

### Testing

1. Deployment
    ```sh
    helm upgrade --install cloudzero-agent . -f override.yaml
    Release "cloudzero-agent" has been upgraded. Happy Helming!
    NAME: cloudzero-agent
    LAST DEPLOYED: Mon Dec 16 17:09:54 2024
    NAMESPACE: default
    STATUS: deployed
    REVISION: 6
    TEST SUITE: None
    ```

2. Inspect deployment
    ````yaml
    apiVersion: apps/v1
     kind: Deployment
    metadata:
      annotations:
        deployment.kubernetes.io/revision: "3"
        meta.helm.sh/release-name: cloudzero-agent
        meta.helm.sh/release-namespace: default
      creationTimestamp: "2024-12-16T19:58:05Z"
      generation: 3
      labels:
        app.kubernetes.io/component: webhook-server
        app.kubernetes.io/instance: cloudzero-agent
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: cloudzero-agent
        app.kubernetes.io/part-of: cloudzero-agent
        app.kubernetes.io/version: v2.50.1
        helm.sh/chart: cloudzero-agent-1.0.0-beta-5
      name: cloudzero-agent-webhook-server
      namespace: default
      resourceVersion: "1195639"
      uid: ce45c0f5-0c7d-4ed1-9129-13bfbd1a93b1
    spec:
      progressDeadlineSeconds: 600
      replicas: 3
      revisionHistoryLimit: 10
      selector:
        matchLabels:
          app.kubernetes.io/component: webhook-server
          app.kubernetes.io/instance: cloudzero-agent
          app.kubernetes.io/name: cloudzero-agent
      strategy:
        rollingUpdate:
          maxSurge: 25%
          maxUnavailable: 25%
        type: RollingUpdate
      template:
        metadata:
          creationTimestamp: null
          labels:
            app.kubernetes.io/component: webhook-server
            app.kubernetes.io/instance: cloudzero-agent
            app.kubernetes.io/managed-by: Helm
            app.kubernetes.io/name: cloudzero-agent
            app.kubernetes.io/part-of: cloudzero-agent
            app.kubernetes.io/version: v2.50.1
            helm.sh/chart: cloudzero-agent-1.0.0-beta-5
        spec:
          affinity:
            podAntiAffinity:
              preferredDuringSchedulingIgnoredDuringExecution: <--------------------------- CONFIRMED
              - podAffinityTerm:
                  labelSelector:
                    matchLabels:
                      app: webhook-server
                  topologyKey: kubernetes.io/hostname
                weight: 100
          containers:
          - args:
            - -config
            - /etc/cloudzero-agent-insights/server-config.yaml
            command:
            - /app/controller
            image: ghcr.io/cloudzero/cloudzero-insights-controller/cloudzero-insights-controller:0.0.5
    ```

**When using different KSM, validate targets config**
![Screenshot 2024-12-16 at 17 12 05](https://github.com/user-attachments/assets/56fb19d0-62f6-4556-84aa-b24e778441b5)


### Checklist

- [x] I have added documentation for new/changed functionality in this PR
- [x] All active GitHub checks for tests, formatting, and security are passing
- [x] The correct base branch is being used, if not `main`

### CODE REVIEW

File: charts/cloudzero-agent/templates/insights-deploy.yaml [hunk 1]
```diff
@@ -37,6 +37,15 @@ spec:
         runAsNonRoot: true
         runAsGroup: 65534
         fsGroup: 65534
+      affinity:
```
Comment (roberthocking) [inline charts/cloudzero-agent/templates/insights-deploy.yaml:40]: This is to make sure we don't schedule all pods on the same node right?

**Review by roberthocking — APPROVED** [review]
Looks good, just one nit

**Review by wreckedred — APPROVED** [review]
Thank you - lgtm

File: charts/cloudzero-agent/templates/insights-deploy.yaml (hunk 1 above)
Comment (josephbarnett) [reply charts/cloudzero-agent/templates/insights-deploy.yaml:40]: Yes - feedback from DVAG was we should add this... to ensure the replicaset it started on different nodes for each pod (if possible)

**Pull request merged**